[9.0][MIG] Upgrade medical to 9.0.

woodbrettm opened the pull request:
Man this diff is absolutely massive. The OCA commit history goes up to Jan 30 2016, and any commits between there and now on release/9.0 are all laslabs ones which is nice. Means I can just copy paste the folder.

**Couple things to bring to light:**
- Are the file headers for python and xml correct? Will be using that header on all files for now on if good.
- Is this [help](https://github.com/laslabs/vertical-medical/compare/9.0...laslabs:custom/9.0/LABS-287-medical-v9?expand=1#diff-0180c4fc21f7260dcce8093bf66149ceR47) attribute correct for is_work?
- Does laslabs/9.0 share the same config files as OCA/9.0? I noticed the travis build had less build jobs than usual (compared to release/9.0), so not sure if the travis config file needs to be improved to add stuff.

Commit:
[MIG] Upgrade medical to 9.0.

lasley commented on medical/__openerp__.py:
```diff
@@ -1,72 +1,30 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Remove our perpetual copyright, also I accidentally deleted some of the original attributions - we will need to fix that.

```
Copyright 2004-2009 Tiny SPRL
Copyright 2016 LasLabs Inc.
```

woodbrettm replied:
Was actually me ;). Wasn't quite sure how to go about that so know now for future.

woodbrettm replied:
Done

lasley commented on medical/README.rst:
```diff
@@ -6,82 +6,64 @@
 
-What is Odoo Medical
---------------------
+This module extends Odoo with the base functionality of medical patients, centers, and doctors.
```
This module does not provide doctors anymore.

woodbrettm replied:
Done

lasley commented on medical/README.rst:
```diff
@@ -6,82 +6,64 @@
 
-**Odoo Medical** is a multi-user, highly scalable, centralized Electronic Medical
-Record (EMR) and Hospital Information System for odoo.
+**Some patient attributes available:**
```
While we can leave it, it's not necessary to list attributes provided. Concept of patient and medical centers does it. If anything, explain what a medical center or patient is, not what their data looks like.

woodbrettm replied:
Reduced to "This module extends Odoo with the base functionality of medical patients and centers." Removed attribute descriptions.

lasley commented on medical/README.rst:
```diff
@@ -6,82 +6,64 @@
 If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
-`here <https://github.com/OCA/vertical-medical/issues/new?body=module:%20medical%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
-
+`here <https://github.com/OCA/vertical-medical/issues/new?body=module:%20medical%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Change this to bug tracker blurb to current template

woodbrettm replied:
Done

lasley commented on medical/__init__.py:
```diff
@@ -1,25 +1,6 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Remove perpetual copyrihgt, add 2004 from Tech-Receptives

woodbrettm replied:
Done

lasley commented on medical/__openerp__.py:
```diff
@@ -1,72 +1,30 @@
+    "version": "9.0.1.0.0",
+    "category": "Medical",
+    "website": "https://odoo-community.org/",
+    "author": "LasLabs, Odoo Medical Team, Odoo Community Association (OCA)",
```
Remove `Odoo Medical Team` - they're not a thing anymore

woodbrettm replied:
Done

lasley commented on medical/__openerp__.py:
```diff
@@ -1,72 +1,30 @@
-        'base',
-        'product',
+    "name": "Odoo Medical",
+    "summary": "Extends Odoo with medical patients, doctors, and centers.",
```
No doctors

woodbrettm replied:
Done

lasley commented on medical/data/ir_sequence_data.xml:
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Remove perpetual copyright

woodbrettm replied:
Done

lasley commented on medical/demo/medical_patient_demo.xml:
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Perpetual copyright

woodbrettm replied:
Done

lasley commented on medical/demo/medical_patient_demo.xml:
```diff
@@ -0,0 +1,58 @@
+<odoo>
+
+	<record id="res_partner_patient_1" model="res.partner">
+		<field name="name">Emma Fields</field>
```
Add ref https://fakena.me/fake-social-security-number/

woodbrettm replied:
Done

lasley commented on medical/demo/medical_patient_demo.xml:
```diff
@@ -0,0 +1,58 @@
+	</record>
+	
+	<record id="res_partner_patient_2" model="res.partner">
+		<field name="name">Kyle Jones</field>
```
Add ref https://fakena.me/fake-social-security-number/

woodbrettm replied:
Done

lasley commented on medical/demo/medical_patient_demo.xml:
```diff
@@ -0,0 +1,58 @@
+	</record>
+	
+	<record id="res_partner_patient_3" model="res.partner">
+		<field name="name">Gary Smith</field>
```
Add ref https://fakena.me/fake-social-security-number/

woodbrettm replied:
Done

lasley commented on medical/demo/medical_patient_demo.xml:
```diff
@@ -0,0 +1,58 @@
+	
+	<record id="medical_patient_patient_3" model="medical.patient">
+		<field name="partner_id" ref="medical.res_partner_patient_3" />
+		<field name="dob">1980-11-02</field>
```
let's kill this patient - add `dod`

woodbrettm replied:
Done

lasley commented on medical/medical_constants.py:
```diff
@@ -1,24 +1,6 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Perpetual copyright, add tech-receptives

woodbrettm replied:
Done

lasley commented on medical/models/__init__.py:
```diff
@@ -1,28 +1,6 @@
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Perpetual copyright, add tech-receptives

woodbrettm replied:
Done

lasley commented on medical/models/medical_patient.py:
```diff
@@ -1,105 +1,108 @@
-#
-# #############################################################################
-from openerp import models, fields, api
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Perpetual copyright, add tech-receptives

woodbrettm replied:
Done

lasley commented on medical/models/medical_patient.py:
```diff
@@ -1,105 +1,108 @@
+                years_months_days = _('No DoB !')
+            rec_id.age = years_months_days
+
+    @api.constrains('is_pregnant', 'gender')
```
Add `api.multi`

woodbrettm replied:
Done. multi placed first then constrains.

lasley commented:
@bmw95 - dunno about `is_work` - just leave it, will probably remove in v10. Don't worry about anything outside of the module dir.

Commit:
[FIX] medical: Fix file headers, add social insurance numbers to patients plus some other small additions and changes.

Commit:
[IMP] medical: Refactor tests to use demo data, add api.depends to some medical_patient model compute methods.

woodbrettm commented:
### Changes Made In Addition to Review Requests
-   Add gender to demo patients
-   Refactor tests to use demo data (so much faster now!)
-   Add patient compute_deceased method (with depends on dod)
-   Add tests that cover the following cases:
  - Ensure pregnant female doesn’t raise any ValidationErrors
  - Split age test into 3 separate tests (no dob, deceased, and normal cases)
  - Test deceased is True if dod
  - Test deceased is False if no dod
  - Refactored action_invalidate test

### Thoughts
- Should there be language translations on strings used in the tests if they can be translated in the core logic (e.g. validationerror messages, etc)?
- Was curious if we should add deceased field to the patient view with logic to allow it to be editable. Would allow for rare cases where the patient is deceased but no dod has been confirmed yet.

Commit:
[FIX] medical: Change manifest website key from oca website to laslabs.com

woodbrettm commented:
Forgot to change manifest website key to laslabs domain.

lasley reviewed (approved):
This looks good to me, thanks @bmw95 

Regarding your questions:
- Don't worry about the translations, non-issue
- There will never be an instance in the medical field of a non-confirmed date of death. If they are marked as dead, a date will be known
